Exercise_ToDoList/js/functions.js

```diff
@@ -1,49 +1,69 @@
+$(document).ready(function ()  {
 
-var taskArray = [
-	{
-		name: "tarefa1",
-		date: "data1"
-	}
+  var taskArray = [
+    {
+      name: "tarefa1",
+      date: "data1"
+    }
+  ]
 
+  showTask()
 
-] 
+  $("#form").on("submit", function() {
 
-function CreatNewTask() {
+    var value1 = $("#task_name").val()
+    var value2 = $("#due_date").val()
 
-   var value1 = document.getElementById('task_name').value
-   var value2 = document.getElementById('due_date').value
- 
-   if (isNaN(value1)) {
 
-      var task = {
-       name: value1,
-       date: value2
-      }
-    
-      taskArray.push(task)
+    if (isNaN(value1)) {
 
-      document.getElementById('task_name').value = ""
-      document.getElementById('due_date').value = ""
-      showTask()
-   }
+        var task = {
+         name: value1,
+         date: value2
+        }
+        // coloca os novos valores no final do vetor
+        taskArray.push(task)
 
-   
-  return false
-}
+        $("#task_name").val("")
+        $("#due_date").val("")
 
-function showTask() {
+        showTask()
+     }
+
+    return false
+  })
+
+  $(".done").click(function() {
+    debugger
+    swal({
+      title: "Good Job",
+      text: "You have concluded another one!",
+      timer: 2000,
+      showConfirmButton: false
+    })
+
+
+
+
+  })
+
+  function showTask() {
+
+    var taskList = ""
+    for (var i = 0; i < taskArray.length; i++) {
```

Comment:
Vale a pena a gente começar a usar javascript funcional para trabalhar com listas. Neste caso, você poderia usar um `forEach`.

ao invés do for, você poderia fazer assim:

```
taskArray.forEach(function (element) {
  taskList += "<tr>" + "<td>" + element.name + ....... + element.date + .....
})
```

poderíamos até usar um reduce. Se conseguir, use um reduce, se não, use um forEach, beleza?